### ISSUE TITLE
Time Series panel: not showing datapoints with JSON datasource after updating to Grafana 8.3.4

### ISSUE THREAD

**Issue by Tilana** [grafana/grafana#44639]
**What happened**:

After updating from Grafana 8.0.4 to Grafana 8.3.4 the Time series panel does not display any data points, only the legend values.
Switching to Table View confirms that the data is there. As a datasource the [JSON plugin](https://grafana.com/grafana/plugins/simpod-json-datasource/) is used.

For Graph (old) the data is displayed without any problems.

Time Series Table View (working):
<img width="1350" alt="Bildschirmfoto 2022-01-31 um 10 34 29" src="https://user-images.githubusercontent.com/10959995/151770233-23cfd488-ab61-41cf-a474-15e39fe7dec1.png">

Time Series Graph View (not working):
<img width="1137" alt="Bildschirmfoto 2022-01-31 um 10 34 45" src="https://user-images.githubusercontent.com/10959995/151770243-8c8160ec-3aa1-4001-925b-c44082d3ee02.png">

Graph (old) (working):
<img width="1100" alt="Bildschirmfoto 2022-01-31 um 10 35 07" src="https://user-images.githubusercontent.com/10959995/151770252-12da1271-2cb4-4a64-a3cc-324f75434427.png">

**What you expected to happen**:

After the Grafana update the Time series panel should show JSON data as before

**How to reproduce it (as minimally and precisely as possible)**:

Mini JSON data to reproduce the error (needs the JSON Plugin setup):

`[{'target': 'Visits', 'datapoints': [[29, datetime.datetime(2022, 1, 1, 3, 0, tzinfo=<UTC>)], [12, datetime.datetime(2022, 1, 2, 3, 0, tzinfo=<UTC>)], [52, datetime.datetime(2022, 1, 3, 3, 0, tzinfo=<UTC>)], [97, datetime.datetime(2022, 1, 4, 3, 0, tzinfo=<UTC>)], [15, datetime.datetime(2022, 1, 5, 3, 0, tzinfo=<UTC>)]]}, {'target': 'base_abs', 'datapoints': [[40, datetime.datetime(2022, 1, 1, 3, 0, tzinfo=<UTC>)], [46, datetime.datetime(2022, 1, 2, 3, 0, tzinfo=<UTC>)], [76, datetime.datetime(2022, 1, 3, 3, 0, tzinfo=<UTC>)], [39, datetime.datetime(2022, 1, 4, 3, 0, tzinfo=<UTC>)], [9, datetime.datetime(2022, 1, 5, 3, 0, tzinfo=<UTC>)]]}]`

**Environment**:
- Grafana version: 8.3.4
- Data source type & version: JSON plugin
- OS Grafana is installed on: Grafana Dockerfile
- User OS & Browser: MacOS & Firefox
- Grafana plugins: JSON plugin
- Others:

**Comment by zuchka** [comment]
Thanks for creating this issue, @Tilana 

I've tried mocking this up but your JSON data does not appear valid on my side. Can you please share your raw request and response? You can find this by clicking your panel title and choosing `inspect` from the dropdown. And then choose `query`, refresh the query, and the copy it to clipboard and paste it here inside triple backticks for a code block ( ``` ).

It might also be useful if you used the inspect feature to share your raw, unformatted data as well 👍

**Comment by Tilana** [comment]
Hi @zuchka.
Thanks for the reply. Below you find the raw query data.
I hope this helps. If any other information is missing, just let know.

```
{
  "request": {
    "url": "api/datasources/proxy/3/query",
    "data": {
      "app": "dashboard",
      "requestId": "Q143",
      "timezone": "browser",
      "panelId": 2,
      "dashboardId": null,
      "range": {
        "from": "2022-01-27T06:55:53.077Z",
        "to": "2022-02-03T06:55:53.077Z",
        "raw": {
          "from": "now-7d",
          "to": "now"
        }
      },
      "timeInfo": "",
      "interval": "10m",
      "intervalMs": 600000,
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "uid": "PDABA066BFC27CFA7",
            "type": "simpod-json-datasource"
          },
          "payload": "",
          "target": "tdi_total_visits_per_day_last7days"
        }
      ],
      "maxDataPoints": 841,
      "scopedVars": {
        "__interval": {
          "text": "10m",
          "value": "10m"
        },
        "__interval_ms": {
          "text": "600000",
          "value": 600000
        }
      },
      "startTime": 1643871353078,
      "rangeRaw": {
        "from": "now-7d",
        "to": "now"
      },
      "adhocFilters": [],
      "endTime": 1643871353102
    },
    "method": "POST",
    "withCredentials": false,
    "hideFromInspector": false
  },
  "response": [
    {
      "datapoints": [
        [
          36,
          "Fri, 28 Jan 2022 02:00:00 GMT"
        ],
        [
          100,
          "Sat, 29 Jan 2022 02:00:00 GMT"
        ],
        [
          94,
          "Sun, 30 Jan 2022 02:00:00 GMT"
        ],
        [
          71,
          "Mon, 31 Jan 2022 02:00:00 GMT"
        ],
        [
          47,
          "Tue, 01 Feb 2022 02:00:00 GMT"
        ],
        [
          62,
          "Wed, 02 Feb 2022 02:00:00 GMT"
        ]
      ],
      "target": "Visits"
    },
    {
      "datapoints": [
        [
          78,
          "Fri, 28 Jan 2022 02:00:00 GMT"
        ],
        [
          6,
          "Sat, 29 Jan 2022 02:00:00 GMT"
        ],
        [
          17,
          "Sun, 30 Jan 2022 02:00:00 GMT"
        ],
        [
          74,
          "Mon, 31 Jan 2022 02:00:00 GMT"
        ],
        [
          83,
          "Tue, 01 Feb 2022 02:00:00 GMT"
        ],
        [
          47,
          "Wed, 02 Feb 2022 02:00:00 GMT"
        ]
      ],
      "target": "base_abs"
    }
  ]
}
```

**Comment by zuchka** [comment]
@Tilana 

So I have some answers / feedback:

1. I was able to mock up this json data inside a test JSON api server and then successfully graph it on 8.3.4 and 8.4.0-pre
2. BUT I didn't use the SimPod Json Plugin that you are using. That is a third-party plugin
3. I would suggest using the [JSON API](https://grafana.com/grafana/plugins/marcusolsson-json-datasource/) plugin instead. I much prefer it and it is getting folded into Grafana proper.
4. If you keep using the SimPod JSON plugin, I would suggest making an issue in their GitHub repo. They are a very active plugin developer
5. Below are a screenshot and my panel JSON. I hope this helps!
6. also, https://community.grafana.com/ is a great place for additional query-help

![CleanShot 2022-02-04 at 14 02 11@2x](https://user-images.githubusercontent.com/37156449/152619151-07bb3139-cd1a-486a-8151-43bca8b737a2.png)

```
{
  "id": 3,
  "gridPos": {
    "h": 9,
    "w": 12,
    "x": 12,
    "y": 0
  },
  "type": "timeseries",
  "title": "Panel Title",
  "transformations": [],
  "datasource": {
    "uid": "Cf0cHV-nk",
    "type": "marcusolsson-json-datasource"
  },
  "pluginVersion": "8.2.2",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "drawStyle": "line",
        "lineInterpolation": "linear",
        "barAlignment": 0,
        "lineWidth": 1,
        "fillOpacity": 0,
        "gradientMode": "none",
        "spanNulls": false,
        "showPoints": "auto",
        "pointSize": 5,
        "stacking": {
          "mode": "none",
          "group": "A"
        },
        "axisPlacement": "auto",
        "axisLabel": "",
        "scaleDistribution": {
          "type": "linear"
        },
        "hideFrom": {
          "tooltip": false,
          "viz": false,
          "legend": false
        },
        "thresholdsStyle": {
          "mode": "off"
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          },
          {
            "color": "red",
            "value": 80
          }
        ]
      },
      "mappings": []
    },
    "overrides": []
  },
  "options": {
    "tooltip": {
      "mode": "single",
      "sort": "none"
    },
    "legend": {
      "displayMode": "list",
      "placement": "bottom",
      "calcs": []
    }
  },
  "targets": [
    {
      "cacheDurationSeconds": 300,
      "fields": [
        {
          "jsonPath": "$.0.datapoints[*].1",
          "type": "time"
        },
        {
          "jsonPath": "$.0.datapoints[*].0",
          "language": "jsonpath",
          "name": ""
        }
      ],
      "method": "GET",
      "queryParams": "",
      "refId": "A",
      "urlPath": "/response"
    },
    {
      "cacheDurationSeconds": 300,
      "fields": [
        {
          "jsonPath": "$.1.datapoints[*].1",
          "type": "time"
        },
        {
          "jsonPath": "$.1.datapoints[*].0",
          "language": "jsonpath",
          "name": ""
        }
      ],
      "hide": false,
      "method": "GET",
      "queryParams": "",
      "refId": "B",
      "urlPath": "/response"
    }
  ]
}
```

**Comment by Tilana** [comment]
Ok, good to know!
Thanks for your time in looking at this issue!

**Comment by Tilana** [comment]
**For the sake of completeness:**
The issue seems to be related with the date format. While I was using a timestamp, e.g. "Wed, 02 Feb 2022 02:00:00 GMT", which works for Graph (old), for the timeseries panel the timestamp in milliseconds is required. 🎉

**Comment by zuchka** [comment]
@Tilana glad you're unblocked! If you ever feel like you are importing a timestamp, but Grafana isn't recognizing it, you can try to use the `Convert field data type` transformation. It works quite well:

https://grafana.com/docs/grafana/latest/panels/transformations/types-options/#convert-field-type

**Comment by dk-gitiot** [comment]
Hi @zuchka and @Tilana  I am trying the same but couldnt display the timeseries data. Im currently using grafana 8.5.3
